addons/portal/models/mail_thread.py

```diff
@@ -87,3 +88,29 @@ def _portal_get_parent_hash_token(self, pid):
         :return: False or logical parent's _sign_token() result
         """
         return False
+
+    @api.model
+    def _get_thread_with_access(self, thread_id, **kwargs):
+        try:
+            thread = super()._get_thread_with_access(thread_id, **kwargs)
+            # Early check on access rights to prevent non-necessary sudo
+            thread.check_access_rights("write")
+            return thread
```

Comment:
as discussed, move check to parent (with a param to config it), only check other stuff if no thread is returned